File: Fideltodayy/reverse_integer.py
def reverse_integer(n):
    """
    Reverses the digits of an integer.

    Parameters:
    n (int): The integer to be reversed.

    Returns:
    int: The reversed integer.
    """
    # Step 1: Handle the sign
    sign = -1 if n < 0 else 1
    abs_number = abs(n)

    # Step 2: Convert to string and reverse
    reversed_str = str(abs_number)[::-1]

    # Step 3: Convert back to integer (this removes leading zeros)
    reversed_number = int(reversed_str)

    # Step 4: Reapply the sign
    return sign * reversed_number

# Example usage:
if __name__ == "__main__":
    # Test cases
    test_inputs = [500, -56, -90, 91, 0, 0000]
    for num in test_inputs:
        reversed_num = reverse_integer(num)
        print(f"Input: {num} → Output: {reversed_num}")

File: r-owino/dsa-clinic/src/entrance-challenge/src/reverse_integer.rs
/*Reverses an integer*/

pub struct Solution;

impl Solution {
    pub fn reverse_integer(x: i32) -> i32 {
        let mut res: i32 = 0;
        let sign = if x < 0 { -1 } else { 1 };
        let mut x = x.abs();

        while x != 0 {
            let remainder = x % 10;
            res = res * 10 + remainder;
            x /= 10;
        }

        res * sign
    }
}
